docs/curl/datatx.md

```diff
@@ -1,213 +1,412 @@
+		]
 	},
-    "signature": "aGVsbG8="
+    "signatures": {
+        "alice": "MEQCIATEMJZ2HYkQtG+ivADylvJRzaksTum3/jN0zeg96+CuAiAEYKugmTbPbHXsjKnAWOLirNqI0WWOPcLN9jIlVaeseQ=="
+    }
 }'
 ```
-Once the above transaction gets committed, we can query `key1` as follows but we would not find any associated value:
+The transaction should be submitted by `alice` as only she has the write permission which is required for the delete operation too.
+The `db_operations` holds the `data_reads` with `key1` and version `"block_num":5`. The `data_deletes` list contains the key `key1`.
+The signature is computed using the following command
+```sh
+./bin/signer -privatekey=deployment/sample/crypto/alice/alice.key -data='{"must_sign_user_ids":["alice"],"tx_id":"1b6d6414-9b58-45d0-9723-1f31712add85","db_operations":[{"db_name":"db2","data_reads":[{"key":"key1","version":{"block_num":6}}],"data_deletes":[{"key":"key1"}]}]}'
+```
+**Output**
+```
+MEQCIATEMJZ2HYkQtG+ivADylvJRzaksTum3/jN0zeg96+CuAiAEYKugmTbPbHXsjKnAWOLirNqI0WWOPcLN9jIlVaeseQ==
+```
+Once the above transaction gets committed, the submitting user would get the following transaction receipt
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "receipt": {
+      "header": {
+        "base_header": {
+          "number": 7,
+          "previous_base_header_hash": "E1QGMExoA6Nt2MEk9GBE+NHDCzpVR4YuBBeYMZONbKU=",
+          "last_committed_block_hash": "WwuMQlvM05Kd/jh+6iD0Kt8ZxMXEdC1wEOk0wzxCBfc=",
+          "last_committed_block_num": 6
+        },
+        "skipchain_hashes": [
+          "WwuMQlvM05Kd/jh+6iD0Kt8ZxMXEdC1wEOk0wzxCBfc=",
+          "xrRU2Hy5curlSi/YafvCCeKaAb8bF9vUNGhr5usK4fc="
+        ],
+        "tx_merkel_tree_root_hash": "vobaBbri1K+lJv+rXYXRofWuQ0D+SDMGKaBwkTtXQQ0=",
+        "state_merkel_tree_root_hash": "jYpTKUQBA7dJZYlzz7gEZWqX503EqGC+z+9pOKJsDKQ=",
+        "validation_info": [
+          {}
+        ]
+      }
+    }
+  },
+  "signature": "MEYCIQDgtHEC+7gaOGsOQbFWOqeCQW5yOrwgDsZI+aZnGu+sogIhAPv+Emp+WxHjUl0ZoSw9mpF50udpTUyI+YqnRXnvn+UA"
+}
+```
+
+### Checking the non-existance of the deleted key
+
+```sh
+./bin/signer -privatekey=deployment/sample/crypto/alice/alice.key -data='{"user_id":"alice","db_name":"db2","key":"key1"}'
+```
+```
+MEUCIQCQWTeWyDwbj2ARFHQTH3pgsplOTrpW6XaC8mC/TODaJgIgIplpD5DB/SSNmhh42qLpITCJc66Z75mcHP3AWsS3zcw=
+```
 ```sh
 curl \
      -H "Content-Type: application/json" \
-     -H "UserID: user1" \
-     -H "Signature: abcd" \
-     -X GET http://127.0.0.1:6001/data/bdb/key1 | jq .
+     -H "UserID: alice" \
+     -H "Signature: MEUCIQCQWTeWyDwbj2ARFHQTH3pgsplOTrpW6XaC8mC/TODaJgIgIplpD5DB/SSNmhh42qLpITCJc66Z75mcHP3AWsS3zcw=" \
+     -X GET http://127.0.0.1:6001/data/db2/key1 | jq .
 ```
 **Output**
 ```json
 {
-  "payload": {
+  "response": {
     "header": {
       "nodeID": "bdb-node-1"
     }
   }
+  "signature": "MEYCIQCVdcbqmwIQDLkwYiK2tmOMwHI0GbWFX6kMMPo4VBORpgIhAMBBvsjCjL8LkogH06m58KefvnMOncLy7uFobh4XNNvI"
 }
```

Comment:
shouldn't this result in a 404?

Reply:
Yes. I think when the provided URL is an invalid endpoint (i.e., mux does not have the URL and method mapping), it would return 404. Anyway, we still need to return `not-found` via `response` payload. I have created this issue to fix the server response https://github.com/IBM-Blockchain/bcdb-server/issues/110